zetaclient/chains/bitcoin/observer/event.go

```diff
@@ -115,11 +116,14 @@ func (event *BTCInboundEvent) DecodeMemoBytes(chainID int64) error {
 		event.MemoStd = memoStd
 		receiver = memoStd.Receiver
 	} else {
-		parsedAddress, _, err := memo.DecodeLegacyMemoHex(hex.EncodeToString(event.MemoBytes))
+		parsedAddress, payload, err := memo.DecodeLegacyMemoHex(hex.EncodeToString(event.MemoBytes))
 		if err != nil { // unreachable code
 			return errors.Wrap(err, "invalid legacy memo")
 		}
 		receiver = parsedAddress
+
+		// update the memo bytes to only contain the data
+		event.MemoBytes = payload
```

Comment:
to confirm, this is due to switching to V2 logic in `zetacore`?
```
else if cctx.ProtocolContractVersion == types.ProtocolContractVersion_V2 {
	if len(cctx.RelayedMessage) > 0 {
		message, err = hex.DecodeString(cctx.RelayedMessage)
		if err != nil {
			return false, errors.Wrap(types.ErrUnableToDecodeMessageString, err.Error())
		}
	}
}
```

Reply:
Yes